### PR TITLE
[QP] Native palette support for fonts

### PR DESCRIPTION
**Pull request by elpekenin**
## Description

PR for native colors on assets didn't update the code for rendering texts, only images, causing some lock ups. To prevent code duplication (and similar issues in the future) i've moved the code into a new function which wraps the two appenders(pixel and byte variants).

I don't like the chosen name very much, im open to suggestions

PS: Can fonts be coloured anyway? 

## Types of Changes

- [X] Core
- [X] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #20632

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/painter/qp_draw_text.c [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // Copyright 2021 Nick Brassel (@tzarc)
+// Copyright 2023 Pablo Martinez (@elpekenin) <elpekenin@elpekenin.dev>
```
Comment (zvecr) [inline quantum/painter/qp_draw_text.c:2]: You're not really going to be able to claim copyright on such a minor change
```suggestion
```

File: quantum/painter/qp_draw_image.c [hunk 2]
```diff
@@ -1,4 +1,5 @@
 // Copyright 2021-2023 Nick Brassel (@tzarc)
+// Copyright 2023 Pablo Martinez (@elpekenin) <elpekenin@elpekenin.dev>
```
Comment (zvecr) [inline quantum/painter/qp_draw_image.c:2]:
```suggestion
```

File: quantum/painter/qp_draw.h [hunk 3]
```diff
@@ -1,4 +1,5 @@
 // Copyright 2021 Nick Brassel (@tzarc)
+// Copyright 2023 Pablo Martinez (@elpekenin) <elpekenin@elpekenin.dev>
```
Comment (zvecr) [inline quantum/painter/qp_draw.h:2]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: quantum/painter/qp_draw_text.c (hunk 1 above)
Comment (elpekenin) [reply quantum/painter/qp_draw_text.c]: > You're not really going to be able to claim copyright on such a minor change

Makes  complete sense and I actually dont mind being on the header or not. For future reference, is there any rule of thumb to know when it should be added?

Comment (zvecr) [reply quantum/painter/qp_draw_text.c]: Realistically for this project and the submissions we get, almost never. There is a case when the majority of a file has been rewritten.

**Comment by tzarc** [comment]
> PS: Can fonts be coloured anyway?

Yes, can create a font image however you like, that's why it's a 2-stage process.

**Comment by elpekenin** [comment]
For what is worth.. i've been running this for a couple months and i didnt notice any issues drawing either images(non-native ones) or fonts(both native and not).

Will try and use a native image tomorrow for the sake of completeness and report back (if i dont forget to test)

**Comment by elpekenin** [comment]
Just confirmed native image too. Replaced an image from `mono2` to `rgb565` and i dont see any difference.

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**